kaiax/builder/impl/tx_pool_wrapping_test.go

```diff
@@ -0,0 +1,624 @@
+			testTxs := map[uint64]*types.Transaction{
+				0: createTestTransaction(0),
+			}
+
+			// Set up mock expectations
+			mockTxBundlingModule.EXPECT().IsBundleTx(gomock.Any()).Return(false).AnyTimes()
+
+			if tt.hasTxPool {
+				mockTxPoolModule := mock_kaiax.NewMockTxPoolModule(ctrl)
+				mockTxPoolModule.EXPECT().IsReady(testTxs, uint64(0), nil).Return(tt.isReadyResult)
+				builderModule.txPoolModule = mockTxPoolModule
+			}
+
+			result := builderModule.IsReady(testTxs, 0, nil)
+			assert.Equal(t, tt.expectedResult, result)
+		})
+	}
+}
+
+func TestPreReset_Timeout(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	now := time.Now()
+
+	tests := []struct {
+		name                 string
+		existingBundles      map[common.Hash]txAndTime
+		expectedBundles      map[common.Hash]txAndTime
+		expectedUnexecutable bool
+	}{
+		{
+			name: "Bundle within timeout period",
+			existingBundles: map[common.Hash]txAndTime{
+				createTestTransaction(0).Hash(): {
+					tx:   createTestTransaction(0),
+					time: now,
+				},
+			},
+			expectedBundles: map[common.Hash]txAndTime{
+				createTestTransaction(0).Hash(): {
+					tx:   createTestTransaction(0),
+					time: now,
+				},
+			},
+			expectedUnexecutable: false,
+		},
+		{
+			name: "Bundle exceeds timeout period",
+			existingBundles: map[common.Hash]txAndTime{
+				createTestTransaction(0).Hash(): {
+					tx:   createTestTransaction(0),
+					time: now.Add(-PendingTimeout - time.Second),
+				},
+			},
+			expectedBundles: map[common.Hash]txAndTime{
+				createTestTransaction(0).Hash(): {
+					tx:   createTestTransaction(0),
+					time: now.Add(-PendingTimeout - time.Second),
+				},
+			},
+			expectedUnexecutable: true,
+		},
+		{
+			name: "Bundle exceeds lock period",
+			existingBundles: map[common.Hash]txAndTime{
+				createTestTransaction(0).Hash(): {
+					tx:   createTestTransaction(0),
+					time: now.Add(-KnownTxTimeout - time.Second),
+				},
+			},
+			expectedBundles:      map[common.Hash]txAndTime{},
+			expectedUnexecutable: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mockTxBundlingModule := mock_builder.NewMockTxBundlingModule(ctrl)
+
+			// Setup BuilderModule
+			builderModule := &BuilderWrappingModule{
+				txBundlingModule: mockTxBundlingModule,
+				knownTxs:         copyTxAndTimeMap(tt.existingBundles),
+			}
+
+			builderModule.PreReset(nil, nil)
+
+			// Verify the number of bundles
+			assert.Equal(t, len(tt.expectedBundles), len(builderModule.knownTxs))
+
+			// Verify each bundle's state
+			for hash, expected := range tt.expectedBundles {
+				actual, exists := builderModule.knownTxs[hash]
+				assert.True(t, exists)
+				assert.Equal(t, expected.tx.Hash(), actual.tx.Hash())
+				assert.Equal(t, expected.time, actual.time)
+				assert.Equal(t, tt.expectedUnexecutable, actual.tx.IsMarkedUnexecutable())
+			}
+		})
+	}
+}
+
+func TestPreReset_TxPoolModule(t *testing.T) {
```

Comment:
What is this test for..?

Reply:
This tests if PreReset of the child module is called if txPoolModule isn't nil.

Reply:
Ah, then can we use the mock feature to check if called?
```
			mockTxPoolModule := mock_kaiax.NewMockTxPoolModule(ctrl)
			if tt.hasTxPoolModule {
				builderModule.txPoolModule = mockTxPoolModule
				mockTxPoolModule.EXPECT().PreReset(nil, nil).Times(1)
			} else {
				mockTxPoolModule.EXPECT().PreReset(nil, nil).Times(0)
			}

			builderModule.PreReset(nil, nil)
```

Reply:
fixed at https://github.com/kaiachain/kaia/pull/309/commits/2c57c57ddd439e58bebd9950c45ab7bfe5405e45